### PR TITLE
custom error, adapted tests

### LINKED ISSUE #1218: [breaking?] Replace ErrorException for not-implemented
`ErrorException` is too generic for not-implemented interface, `_NI("func")` should either throw a method error or a custom `Exception` type

### PR DESCRIPTION
**Pull request by matbesancon**
Closes #1218

### CODE REVIEW

**Comment by matbesancon** [comment]
Note I didn't export the error, this is on purpose:
- I think `LightGraphs.ImplementationError` reads better than `LightGraphs.ImplementationError`
- Think that there might be several packages and ecosystems the user is manipulating at the same time, so potentially several "interface implementation error" sources

File: test/persistence/persistence.jl [hunk 1]
```diff
@@ -1,7 +1,4 @@
 @testset "Persistence" begin
-    @testset "Errors" begin
```
Comment (matbesancon) [inline test/persistence/persistence.jl]: @sbromberger the test removed here feels like a "comfort test" more than anything, it's literally just the function behavior

**Comment by matbesancon** [comment]
ping @sbromberger here

**Comment by sbromberger** [comment]
Ping. Can we get that last bit of coverage up? Thanks.

**Comment by matbesancon** [comment]
@sbromberger all good here

**Pull request merged**